controllers/designateapi_controller.go

```diff
@@ -725,43 +726,45 @@ func (r *DesignateAPIReconciler) reconcileNormal(ctx context.Context, instance *
 			condition.DeploymentReadyRunningMessage))
 		return ctrlResult, nil
 	}
-	instance.Status.ReadyCount = depl.GetDeployment().Status.ReadyReplicas
-
-	// verify if network attachment matches expectations
-	networkReady := false
-	networkAttachmentStatus := map[string][]string{}
-	if *(instance.Spec.Replicas) > 0 {
-		networkReady, networkAttachmentStatus, err = nad.VerifyNetworkStatusFromAnnotation(
-			ctx,
-			helper,
-			instance.Spec.NetworkAttachments,
-			serviceLabels,
-			instance.Status.ReadyCount,
-		)
-		if err != nil {
-			return ctrl.Result{}, err
+	if depl.GetDeployment().Generation <= depl.GetDeployment().Status.ObservedGeneration {
```

Comment:
based on the discussion in the docs PR https://github.com/openstack-k8s-operators/dev-docs/pull/102#discussion_r1565989013 I think the < case also means we need to reconcile again instead of declaring success.j